doc/guides/prog_guide/cnet.rst

```diff
@@ -0,0 +1,389 @@
+
+- **punt_kernel** is the node to send packets to the kernel to be processed if the packet is not being processed by the application.
+
+  - Sending these packets to the kernel allows other processes waiting for packets on standard linux sockets to process the packets.
+
+- **arp_request** is the node used to send packets to the Linux kernel stack if the ARP entry is not known by CNET.
+
+  - ARP packet processing is handled by the Linux kernel and not by CNET. When the CNET stack needs a destination MAC address, and it is not known, an ARP request is sent by the Linux stack.
+
+- **chnl_callback** is the node to callback into the application to allow for more packet data processing to continue.
+
+  - When a channel is created a callback routine is given to allow the CNET to callback into the application to help complete the connection or data processing.
+
+Anatomy of CNET processing
+--------------------------
+
+The anatomy of the CNET stack is complex, but most of the processing and configuration is defined by
+the :ref:`Graph Library <Graph_Library>`, but a number of internal structures and designs should be understood.
+One of the design details is the CNET stack uses a couple thread local variables *this_cnet* and *this_stk*
+to help avoid passing these values in the APIs.
+
+The :ref:`CNET Structure <figure_cnet_structure>` is a single global structure containing information about
+all of the :ref:`Stack Structures (stk_t) <figure_cnet_stack_structure>`. Each stack instance is independent of each
+other except for the needed information from the *cnet* structure. The *stk_t* structure contains information
+about each graph instance, which contains a set of graph nodes for the given stack instance. Each graph instance
+can contain different graph nodes. The cnet-graph example always has the same set of graph nodes in each instance.
+An application could have different graph layouts for each instance of a graph, but *cnet-graph* will always
+have the same set of the same graph nodes in each instance of a graph.
+
+.. note:: For the internals of the *cnet* and *stk_t* structures refer to the *Doxygen* created API
+  documentation. The *channel* APIs are also defined in the documentation.
+
+The application uses the Channel APIs to create connections or setup listening connections similar
+to the socket API. Using APIs like *channel()*, *chnl_recv()*, *chnl_send()*, *chnl_bind()*, *chnl_listen()*,
+*chnl_connect()*, *chnl_open()* and *chnl_accept()* are a few of the APIs to use for creating connections. The
+APIs look similar to standard *Socket* APIs.
+
+.. _figure_cnet_stack_view:
+
+.. figure:: img/cnet_stack_view.*
+
+   CNET Stack High Level View
+
+CNET Structure
+^^^^^^^^^^^^^^
+
+The CNET structure :ref:`CNET Structure <figure_cnet_structure>` contains a number of fields to describe
+the information to control the CNET stack. This structure is created once for all stack instances.
+
+.. _figure_cnet_structure:
+
+.. code-block:: c
```

Comment:
I understand, can I leave it here as I wanted to be more descriptive. How about I add a pointer in the header files to the docs?

Reply:
I'm ok leaving it in. I don't think its necessary to add a comment in the actual header files, since you'd have to duplicate it for every structure that you duplicating the documentation for.